apps/studio/src/server/modules/asset/asset.service.ts

```diff
@@ -0,0 +1,21 @@
+import { randomUUID } from "crypto"
+import type { z } from "zod"
+
+import type { getPresignedPutUrlSchema } from "~/schemas/asset"
+import { generateSignedPutUrl } from "~/lib/s3"
+
+export const getFileKey = ({
+  siteId,
+  fileName,
+}: z.infer<typeof getPresignedPutUrlSchema>) => {
+  // NOTE: We're using a random folder name to prevent collisions
+  const folderName = randomUUID()
+
+  return `${siteId}/${folderName}/${fileName}`
```

Comment:
hwy not just use the resource id since it's guaranteed to be unique

Reply:
Hmm this is the asset itself and it's possible to have multiple images within the same page resource, so will need a more globally unique ID.